### PR TITLE
Improve RTL support in HTML-CSS

### PR DESCRIPTION
**Pull request by pkra**
Add CSS to prevent browser interference

### CODE REVIEW

**Comment by pkra** [comment]
Fixes #1250

**Comment by pkra** [comment]
Marking this WIP in the hopes of doing something about #1252. But perhaps that's too optimistic.

**Comment by pkra** [comment]
I would like to add the fix for Edge before merging this. But that can wait until the beta is out.

**Comment by pkra** [comment]
Added a fix for #1252, including browser detection for MS Edge in MathJax.js.

Ready for review.

**Comment by pkra** [comment]
Hold off on that review. A more extensive browser test revealed additional issues on IE.

**Comment by pkra** [comment]
Adding the Edge hack for MSIE might be sufficient for the time being. 

To be precise: using http://codepen.io/pkra/pen/MaYeBZ I see

* IE11 and IE10 
  * all clear 
  * without Edge hack: 1,6,8 get processing errors
* IE9: 
  * failed: 3,4, 7 get processing errors, 6 (stray p node in first part)
  * without Edge hack: 1, 3, 4, 6, 7, 8 get processing errors
* IE8: 
  * failed: 1 (layout: full underline), 6 (processing error in first part), 8 (layout: full underlines) 
  * without Edge hack: 1, 6, 8 get processing errors
* IE7: 
  * 1 (layout: full underline), 6 (stray p node in first part), 8 (layout: full underlines) 
  * without Edge hack: 1, 6, 8 get processing errors

**Comment by pkra** [comment]
I think it's ready for review as the remaining issues seem to be upstream issues. (I suspect the improvements from 2.5 never worked on older IE :-( )

File: unpacked/MathJax.js [hunk 1]
```diff
@@ -3090,6 +3090,7 @@ MathJax.Hub.Startup = {
     isMac:       (navigator.platform.substr(0,3) === "Mac"),
     isPC:        (navigator.platform.substr(0,3) === "Win"),
     isMSIE:      ("ActiveXObject" in window && "clipboardData" in window),
+    isMsEdge:    (!!AGENT.match(/Edge\//)),
```
Comment (dpvc) [inline unpacked/MathJax.js]: I thought you meant that you were going to use `!!AGENT.match(/Edge\//)` in `mml3.js` itself, not modify `MathJax.js`.  The latter requires additional work:  you would need to handle the version number later in the code, and you would also need to make sure that `isSafari` isn't also set (which it still will be).

File: unpacked/extensions/MathML/mml3.js [hunk 2]
```diff
@@ -121,6 +765,9 @@ MathJax.Hub.Register.StartupHook("MathML Jax Ready",function () {
     // No XSLT support. Do not change the <math> content.
     MATHML.mml3XSLT = null;
   }
+  
+  // Tweak CSS to avoid some browsers rearranging HTML output
+  MathJax.Ajax.Styles(".MathJax span { direction: ltr !important; display: inline-block !important;}");
```
Comment (dpvc) [inline unpacked/extensions/MathML/mml3.js]: This is probably over-aggressive, and may cause unforeseen problems, particularly the `display` setting.  I will see if I can suggest a left intrusive set of styles.

File: unpacked/MathJax.js (hunk 1 above)
Comment (pkra) [inline unpacked/MathJax.js]: Ah, sorry. I misunderstood the earlier discussion. Happy to do it in just mml3.js for the time being.

File: unpacked/extensions/MathML/mml3.js (hunk 2 above)
Comment (pkra) [inline unpacked/extensions/MathML/mml3.js]: Thanks, that would be great.

Comment (pkra) [inline unpacked/extensions/MathML/mml3.js]: PS: Feel free to update the branch directly.

**Comment by dpvc** [comment]
@pkra, if you are updating the branch, you should also change `FastHTML` to `PreviewHTML` since that is what we ended up deciding on as the final name for the old output mode.

**Comment by pkra** [comment]
>  you should also change FastHTML to PreviewHTML 

Wrong issue? I think you mean https://github.com/mathjax/MathJax-dev/pull/14, correct?

**Comment by dpvc** [comment]
OOPS, sorry, too many windows open at once.  Yes, that is the correct issue.

**Comment by pkra** [comment]
I wrote earlier:

> I suspect the improvements from 2.5 never worked on older IE :-(

I actually confirmed that later but never reported here. So for the record, there are no regressions as far as I can tell only improvements.

**Pull request merged**